### PR TITLE
[Wallet] Remote notification targeting specific app versions and countries

### LINKED ISSUE #6548: Firebase controlled notifications should be able to target specific app versions and countries
### Expected Behavior

Firebase controlled notifications should be able to target specific app versions and countries.

### Current Behavior

Firebase controlled notifications for the `<NotificationBox />` are displayed on all app versions and countries.
This is a problem when introducing new features that we only want to advertise on apps that support said feature.
Also some features are country specific.

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

This PR allows targeting specific app versions and countries for notifications displayed in the `<NotificationBox />`.

If `minVersion` / `maxVersion` is set, the notification will only be displayed if the app version is in the range.
If `countries` (array of 2 letters country codes, see https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) is set, the notification will only be displayed if the entered phone number belongs to the array.

It depends on #6512 

### Other changes

- Fixed `UPDATE_NOTIFICATIONS` action not clearing previously set fields

### Tested

- Updated tests.
- Manually on Alfajores

Screenshot 2021-01-20 at 15 07 55

### Related issues

- Fixes #6548

### Backwards compatibility

Yes, currently released apps read from `notifications` in Firebase, but releases including this change will read from `notificationsV2`.
This way, existing released apps won't display the new targeted notifications.

### CODE REVIEW

File: packages/mobile/src/home/NotificationBox.tsx [hunk 1]
```diff
@@ -223,7 +224,7 @@ export class NotificationBox extends React.Component<Props, State> {
     }
 
     for (const [id, notification] of Object.entries(this.props.extraNotifications)) {
-      if (!notification || notification.dismissed) {
```
Comment (gnardini) [inline packages/mobile/src/home/NotificationBox.tsx:226]: Is this change intentional?

File: packages/mobile/src/home/reducers.ts [hunk 2]
```diff
@@ -53,11 +56,17 @@ export const homeReducer = (state: State = initialState, action: ActionTypes | R
         if (!updatedNotification) {
           continue
         }
+        const existingNotification = state.notifications[id]
         updatedNotifications = {
           ...updatedNotifications,
           [id]: {
-            ...(state.notifications[id] || {}),
             ...updatedNotification,
+            // Keep locally modified fields
```
Comment (gnardini) [inline packages/mobile/src/home/reducers.ts:64]: This only affects us if we put a value for `dismissed` in Firebase, right? If there's no such field on Firebase, we were still keeping the locally stored value.

If that's the case I'm not sure I see the need for this change, but I don't see anything wrong with it either. Might be missing something though.

**Review by gnardini — COMMENTED** [review]
Awesome to have this additional flexibility! 😄

File: packages/mobile/src/home/NotificationBox.tsx (hunk 1 above)
Comment (jeanregisser) [reply packages/mobile/src/home/NotificationBox.tsx:226]: Yes it's now covered in the selector.

File: packages/mobile/src/home/reducers.ts (hunk 2 above)
Comment (jeanregisser) [reply packages/mobile/src/home/reducers.ts:64]: Yes what I saw happening while testing is that previously set values would remain in the redux store.
For instance if I set `minVersion` to something for testing and then would delete the field in Firebase, it would stay in the redux state. Leading to the notification remaining hidden or visible when it shouldn't according to what's in Firebase.

**Review by gnardini — APPROVED** [review]

**Comment by Lss-Ankit** [comment]
Hi @jeanregisser We have verified this on latest testflight build V1.9.0  & Android internal V1.9.0 and observe the followings.
- Simplex page is shown for Germany
- Spend option is shown when user login with PHP 
- "Valora is not available in your country" message is shown when user taps on Next button from Account info screen after selecting and entering phone number for below countries 

**Banned Country list :-** Iran, Cuba, North Korea, Sudan, Syria

Can you please give us more information about this task for now we have verified above points & please let us know if we need to test anything else.
Thanks!
